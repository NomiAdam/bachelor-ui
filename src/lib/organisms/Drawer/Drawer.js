import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {Grid, GridCol} from '../../atoms/Grid/index';
import NavigationBar from './components/NavigationBar/index';
import ApplicationBar from './components/ApplicationBar/index';

const StyledDrawer = styled.div`
	width: 100%;
	height: 100%;
`;

const StyledContent = styled.div`
	width: 100%;
	height: 100%;
`;

const Drawer = ({children, links, onLogout, falconName, userName}) => (
        <StyledDrawer>
            <Grid>
                <GridCol colXS={12} >
                    <ApplicationBar
                        onLogout={onLogout}
                        falconName={falconName}
                        userName={userName}
                    />
                </GridCol>
            </Grid>
            <Grid>
                <GridCol colXS={3} colMD={2}>
                    <NavigationBar
                        links={links}
                    />
                </GridCol>
                <GridCol colXS={9} colMD={10}>
                    <StyledContent>
                        {children}
                    </StyledContent>
                </GridCol>
            </Grid>
        </StyledDrawer>
);

Drawer.propTypes = {
    /**
     * Displays name of logged falcon
     */
    falconName: PropTypes.string,
    /**
     * Displays name of logged user
     */
    userName: PropTypes.string,
    /**
     * Logout function
     */
    onLogout: PropTypes.func,
    /**
     * Array of React router Links
     */
    links: PropTypes.any,
    /**
     * Content of the application
     */
    children: PropTypes.any,
};

export { StyledDrawer, StyledContent };
export default Drawer;
