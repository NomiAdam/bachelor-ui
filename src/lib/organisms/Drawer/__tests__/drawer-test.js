/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import 'jest-styled-components';
import Drawer, {StyledDrawer} from '../Drawer';

describe('Drawer', () => {
    it('should be defined', () => {
        expect(Drawer).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Drawer/>)).toMatchSnapshot();
    });
    it('should match the snapshot with children', () => {
        expect(shallow(
            <Drawer>
                <p>mock</p>
            </Drawer>
        )).toMatchSnapshot();
    });
    it('should be a StyledDrawer', () => {
        expect(shallow(<Drawer/>).type()).toBe(StyledDrawer);
    });
});