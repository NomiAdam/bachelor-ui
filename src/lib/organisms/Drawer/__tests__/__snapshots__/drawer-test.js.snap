// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should match the snapshot 1`] = `
<styled.div>
  <styled.div
    gridGap="10px"
    padding="10px"
  >
    <styled.div
      colXS={12}
    >
      <ApplicationBar />
    </styled.div>
  </styled.div>
  <styled.div
    gridGap="10px"
    padding="10px"
  >
    <styled.div
      colMD={2}
      colXS={3}
    >
      <NavigationBar />
    </styled.div>
    <styled.div
      colMD={10}
      colXS={9}
    >
      <styled.div />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Drawer should match the snapshot with children 1`] = `
<styled.div>
  <styled.div
    gridGap="10px"
    padding="10px"
  >
    <styled.div
      colXS={12}
    >
      <ApplicationBar />
    </styled.div>
  </styled.div>
  <styled.div
    gridGap="10px"
    padding="10px"
  >
    <styled.div
      colMD={2}
      colXS={3}
    >
      <NavigationBar />
    </styled.div>
    <styled.div
      colMD={10}
      colXS={9}
    >
      <styled.div>
        <p>
          mock
        </p>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
